kpatch-build/create-diff-object.c

```diff
@@ -983,8 +1002,12 @@ static void kpatch_correlate_static_local_variables(struct kpatch_elf *base,
 				continue;
 			}
 
-			patched_sym = kpatch_find_static_twin(sec, sym);
+			patched_sym = kpatch_find_static_twin(toc_sec, sym);
```

Comment:
I think this is still not quite right.  It's comparing the .rela.toc section relas.  Instead it should be comparing the original .rela.data.* section relas.  Otherwise it may get confused and correlate the wrong variables.  For example, if there are multiple static locals with the same name, but in different .data.* sections, they should *not* be correlated.  But this code might accidentally correlate them, because they're both referenced in .rela.toc.

Instead I think the original .data.* section should be passed to kpatch_find_static_twin(), and kpatch_find_static_twin() should be fixed to compare toc_rela-based symbols.

The static local variable correlation is a bit confusing (and actually needs to be simplified).  If you want, I can try to take a pass at fixing this bug.

Reply:
> I think this is still not quite right. It's comparing the .rela.toc section relas. Instead it should be comparing the original .rela.data.* section relas.

This stuff is really hard to communicate precisely, it might be easier if I write a patch to communicate it :-)